frontend/public/components/overview/build-configs-overview.tsx

```diff
@@ -0,0 +1,88 @@
+import * as _ from 'lodash-es';
+import * as React from 'react';
+import { Link } from 'react-router-dom';
+import { Button, ListGroup } from 'patternfly-react';
+
+import { BuildPhaseIcon } from '../build';
+import { errorModal } from '../modals/error-modal';
+import { startBuild } from '../../module/k8s/builds';
+import { fromNow } from '../utils/datetime';
+import { ResourceLink, resourcePath, SidebarSectionHeading } from '../utils';
+
+const conjugateBuildPhase = (phase) => {
+  switch (phase) {
+    case 'Cancelled':
+      return 'was cancelled';
+    case 'Error':
+      return 'encountered an error';
+    case 'Failed':
+      return 'failed';
+    default:
+      return `is ${_.toLower(phase)}`;
+  }
+};
+
+const BuildOverviewItem = ({build}) => {
+  const {metadata} = build;
+  const timestamp = build.status.completionTimestamp || build.status.startTimestamp;
+
+  return <li className="list-group-item build-config-overview__item">
+    <div>
+      <BuildPhaseIcon build={build} />
+      &nbsp;
+      <ResourceLink
+        hideIcon
+        inline
+        kind="Build"
+        name={metadata.name}
+        namespace={metadata.namespace}
+      />
+      {conjugateBuildPhase(build.status.phase)}
+      {timestamp && <span className="text-muted">&nbsp;({fromNow(timestamp)})</span>}
+    </div>
+    <div>
+      <Link to={`${resourcePath('Build', build.metadata.name, build.metadata.namespace)}/logs`}>
+        View Logs
+      </Link>
+    </div>
+  </li>;
+};
+
+const BuildConfigOverviewList: React.SFC<any> = ({buildConfig}) => {
+  const {metadata, builds} = buildConfig;
+  const onClick = () => {
+    startBuild(buildConfig).catch(err => {
+      const error = err.message;
+      errorModal({error});
+    });
+  };
+  return <ListGroup className="build-config-overview__list" componentClass="ul">
+    <li className="list-group-item build build-config-overview__item">
+      <div>
+        <ResourceLink
+          inline
+          kind="BuildConfig"
+          name={metadata.name}
+          namespace={metadata.namespace}
+        />
+      </div>
+      <div>
+        <Button bsStyle="link" onClick={onClick}>Start Build</Button>
+      </div>
+    </li>
+    {
+      _.isEmpty(builds)
+        ? <li className="list-group-item"><span className="text-muted">No Builds found for this Build Config</span></li>
+        : _.map(builds, build => <BuildOverviewItem key={build.metadata.uid} build={build} />)
+    }
+  </ListGroup>;
+};
+
+export const BuildConfigsOverview: React.SFC<any> = ({buildConfigs}) => <div className="build-config-overview">
```

Comment:
Exported components generally should have props types defined. (Ideally the other components above, too.)